network/src/protocols/health_checker/test.rs

```diff
@@ -15,136 +15,148 @@ use crate::{
-    network_reqs_rx: &mut diem_channel::Receiver<(PeerId, ProtocolId), PeerManagerRequest>,
-) {
-    let (_ping_msg, res_tx) = expect_ping(network_reqs_rx).await;
-    // This mock ping request must fail.
-    res_tx.send(Err(RpcError::TimedOut)).unwrap();
-}
+    async fn expect_ping_send_notok(&mut self) {
+        let (_ping_msg, res_tx) = self.expect_ping().await;
+        // This mock ping request must fail.
+        res_tx.send(Err(RpcError::TimedOut)).unwrap();
+    }
 
-async fn expect_ping_timeout(
-    network_reqs_rx: &mut diem_channel::Receiver<(PeerId, ProtocolId), PeerManagerRequest>,
-) {
-    let (_ping_msg, _res_tx) = expect_ping(network_reqs_rx).await;
-    // Sleep for ping timeout plus a little bit.
-    std::thread::sleep(PING_TIMEOUT + Duration::from_millis(100));
-}
+    async fn send_inbound_ping(
+        &mut self,
+        peer_id: PeerId,
+        ping: u32,
+    ) -> oneshot::Receiver<Result<Bytes, RpcError>> {
+        let protocol_id = ProtocolId::HealthCheckerRpc;
+        let data = bcs::to_bytes(&HealthCheckerMsg::Ping(Ping(ping)))
+            .unwrap()
+            .into();
+        let (res_tx, res_rx) = oneshot::channel();
+        let inbound_rpc_req = InboundRpcRequest {
+            protocol_id,
+            data,
+            res_tx,
+        };
+        let key = (peer_id, ProtocolId::HealthCheckerRpc);
+        let (delivered_tx, delivered_rx) = oneshot::channel();
+        self.peer_mgr_notifs_tx
+            .push_with_feedback(
+                key,
+                PeerManagerNotification::RecvRpc(peer_id, inbound_rpc_req),
+                Some(delivered_tx),
+            )
+            .unwrap();
+        delivered_rx.await.unwrap();
+        res_rx
+    }
 
-async fn send_inbound_ping(
-    peer_id: PeerId,
-    ping: u32,
-    network_notifs_tx: &mut diem_channel::Sender<(PeerId, ProtocolId), PeerManagerNotification>,
-) -> oneshot::Receiver<Result<Bytes, RpcError>> {
-    let protocol_id = ProtocolId::HealthCheckerRpc;
-    let data = bcs::to_bytes(&HealthCheckerMsg::Ping(Ping(ping)))
-        .unwrap()
-        .into();
-    let (res_tx, res_rx) = oneshot::channel();
-    let inbound_rpc_req = InboundRpcRequest {
-        protocol_id,
-        data,
-        res_tx,
-    };
-    let key = (peer_id, ProtocolId::HealthCheckerRpc);
-    let (delivered_tx, delivered_rx) = oneshot::channel();
-    network_notifs_tx
-        .push_with_feedback(
-            key,
-            PeerManagerNotification::RecvRpc(peer_id, inbound_rpc_req),
-            Some(delivered_tx),
-        )
-        .unwrap();
-    delivered_rx.await.unwrap();
-    res_rx
+    async fn expect_disconnect(&mut self, expected_peer_id: PeerId) {
+        let req = self.connection_reqs_rx.next().await.unwrap();
+        let (peer_id, res_tx) = match req {
+            ConnectionRequest::DisconnectPeer(peer_id, res_tx) => (peer_id, res_tx),
+            _ => panic!("Unexpected ConnectionRequest: {:?}", req),
+        };
```

Comment:
It's too bad there isn't something built into rust to let you do this, since it's like 3-4 places in here.  Where you expect a certain type, otherwise `panic!`

Reply:
Yep... wish there was a way to pull out a specific enum variant and then panic with a descriptive error message like this...